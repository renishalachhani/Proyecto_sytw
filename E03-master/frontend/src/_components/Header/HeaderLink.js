import { Link } from 'react-router-dom'
import styled from 'styled-components'

export default styled(Link)`
  display: inline-flex;
  padding: 0.25em 2em;
  margin: 1em;
  text-decoration: none;
  user-select: none;
  cursor: pointer;
  font-size: 18px;
  color: black;

  &:active {
    background: transparent;
    color: transparent;
  }
`
