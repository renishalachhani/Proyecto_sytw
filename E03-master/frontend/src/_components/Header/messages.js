/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage component.
 */
import { defineMessages } from 'react-intl'

export const scope = 'boilerplate.components.Header'

export default defineMessages({
  home: {
    id: `${scope}.home`,
    defaultMessage: 'Home',
  },

  SignIn: {
    id: `${scope}.SignIn`,
    defaultMessage: 'Sign In',
  },

  LogOut: {
    id: `${scope}.LogOut`,
    defaultMessage: 'Log Out',
  },

  Profile: {
    id: `${scope}.Profile`,
    defaultMessage: 'Profile',
  },

  SignUp: {
    id: `${scope}.SignUp`,
    defaultMessage: 'Sign Up',
  },

  Category: {
    id: `${scope}.Category`,
    defaultMessage: 'Category',
  },

  CreateEvent: {
    id: `${scope}.CreateEvent`,
    defaultMessage: 'Create event',
  },
})
