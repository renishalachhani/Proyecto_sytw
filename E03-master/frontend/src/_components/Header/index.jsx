import React from 'react'
import { FormattedMessage } from 'react-intl'
import NavBar from './NavBar'
import HeaderLink from './HeaderLink'
import messages from './messages'
import { assets } from '../../_helpers/assets'
import './header.css'

function logged() {
  return localStorage.getItem('user') !== null ? true : false
}

function Header() {
  return (
    <div className="d-flex align-items-center">
      <div className="bd-highlight justify-content-start ml-4 mr-2">
        <img
          src={assets('/images/logo.png')}
          alt="logo"
          className="img-responsive img-fluid"
          height="90"
          width="90"
        />
      </div>

      <div className="p-2 bd-highlight">
        <NavBar>
          {!logged() && (
            <div>
              <HeaderLink to="/login">
                <FormattedMessage {...messages.SignIn} />
              </HeaderLink>

              <HeaderLink to="/register">
                <FormattedMessage {...messages.SignUp} />
              </HeaderLink>
            </div>
          )}

          {logged() && (
            <div>
              <HeaderLink to="/">
                <FormattedMessage {...messages.home} />
              </HeaderLink>

              <HeaderLink to="/profile">
                <FormattedMessage {...messages.Profile} />
              </HeaderLink>

              <HeaderLink to="/create-event">
                <FormattedMessage {...messages.CreateEvent} />
              </HeaderLink>

              <HeaderLink to="/logout" className="logout">
                <FormattedMessage {...messages.LogOut} />
              </HeaderLink>
            </div>
          )}
        </NavBar>
      </div>
    </div>
  )
}

export default Header
