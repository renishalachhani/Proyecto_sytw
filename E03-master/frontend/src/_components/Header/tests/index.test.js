import jsdom from 'jsdom'
import React from 'react'
import { mount, configure, render } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Header from '../index.jsx'
import renderer from 'react-test-renderer'
import { createMemoryHistory } from 'history'
import { Router } from 'react-router'

import LocalStorageMock from "../../../../__mocks__/localStorageMock"

global.localStorage = new LocalStorageMock;

configure({ adapter: new Adapter() })

const { JSDOM } = jsdom
const dom = new JSDOM('<!doctype html><html><body></body></html>')
const { window } = dom

global.window = window
global.document = window.document

var history = createMemoryHistory();

describe('<Header />', () => {
  test('Comprobar que se renderiza Header', () => {
    const wrapper = mount(
      <Router history={history}>
        <Header />
      </Router>
    )
    expect(wrapper.find(Header)).toHaveLength(1)
  })

  it('Snapshot', () => {  
    const Component = renderer.create(
      <Router history={history}>
        <Header />
      </Router>
    ).toJSON();
    expect(Component).toMatchSnapshot();
  });
})
