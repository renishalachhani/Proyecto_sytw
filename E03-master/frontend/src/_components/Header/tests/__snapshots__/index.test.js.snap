// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Snapshot 1`] = `
<div
  className="d-flex align-items-center"
>
  <div
    className="bd-highlight justify-content-start ml-4 mr-2"
  >
    <img
      alt="logo"
      className="img-responsive img-fluid"
      height="90"
      src="http://localhost:4000/images/logo.png"
      width="90"
    />
  </div>
  <div
    className="p-2 bd-highlight"
  >
    <div
      className="sc-bdVaJa jhRYkF"
    >
      <div>
        <a
          className="sc-bwzfXH jPTqTw"
          href="/login"
          onClick={[Function]}
        >
          Sign In
        </a>
        <a
          className="sc-bwzfXH jPTqTw"
          href="/register"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </div>
    </div>
  </div>
</div>
`;
